### PR TITLE
ENH: updating subset types in drop_duplicates/duplicated

### PR DESCRIPTION
**Pull request by kirill-bash**
- [x] Part of #[59237](https://github.com/pandas-dev/pandas/issues/59237) (Replace xxxx with the Github issue number)
- [x] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Comment by kirill-bash** [comment]
Follow-up to https://github.com/pandas-dev/pandas/pull/59392

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Can you add a test for this in `tests/test_frame.py` ?

**Comment by kirill-bash** [comment]
@Dr-Irv,

I have added a test for the supported arguments.

File: tests/test_frame.py [hunk 1]
```diff
@@ -368,6 +368,33 @@ def test_types_dropna() -> None:
...
+    "drop_arg",
+    [
+        {"AAA"},              # set
+        ["AAA"],              # list
+        ("AAA",),             # tuple
+        {"AAA": None},        # dict
+        "AAA",                # str
+    ]
+)
+def test_types_drop_duplicates(drop_arg) -> None:
```
Comment (Dr-Irv) [inline tests/test_frame.py:381]: For the purposes of testing type checks, you have to explicitly put in each of these arguments and can't parameterize.   So you'd do:
```python
check(assert_type(df.drop_duplicates({"AAA"}), pd.DataFrame), pd.DataFrame)
check(assert_type(df.drop_duplicates(["AAA"]), pd.DataFrame), pd.DataFrame)
# and similar for the other 3
```
Then you don't need to do `assert_frame_equal()`

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

**Comment by kirill-bash** [comment]
@Dr-Irv,

Updated based on your feedback.

Thanks!

File: tests/test_frame.py [hunk 2]
```diff
@@ -368,6 +368,24 @@ def test_types_dropna() -> None:
...
+            "C": [1, 1, 2, 2, 2, 2, 1, 2],
+            "D": range(8),
+        }
+    )
+
+    check(assert_type(df.drop_duplicates({"AAA"}), pd.DataFrame), pd.DataFrame)
+    check(assert_type(df.drop_duplicates(["AAA"]), pd.DataFrame), pd.DataFrame)
+    check(assert_type(df.drop_duplicates(("AAA",)), pd.DataFrame), pd.DataFrame)
+    check(assert_type(df.drop_duplicates({"AAA": None}), pd.DataFrame), pd.DataFrame)
+    check(assert_type(df.drop_duplicates("AAA"), pd.DataFrame), pd.DataFrame)
```
[the remark this answers is not in the document]
Comment (kirill-bash) [reply tests/test_frame.py]: Thanks for the tip!

File: tests/test_frame.py [hunk 3]
```diff
@@ -379,12 +379,14 @@ def test_types_drop_duplicates() -> None:
         }
     )
 
-    check(assert_type(df.drop_duplicates({"AAA"}), pd.DataFrame), pd.DataFrame)
     check(assert_type(df.drop_duplicates(["AAA"]), pd.DataFrame), pd.DataFrame)
     check(assert_type(df.drop_duplicates(("AAA",)), pd.DataFrame), pd.DataFrame)
-    check(assert_type(df.drop_duplicates({"AAA": None}), pd.DataFrame), pd.DataFrame)
     check(assert_type(df.drop_duplicates("AAA"), pd.DataFrame), pd.DataFrame)
 
+    if PD_LTE_22:
```
Comment (Dr-Irv) [inline tests/test_frame.py:386]:
```suggestion
    if not PD_LTE_22:
```
You want this test to only be executed if we are in a version after pandas 2.2

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
You need to run black on your changed files, or follow the instructions at https://github.com/pandas-dev/pandas-stubs/blob/main/docs/setup.md to set up your environment and install the pre-commit checks.

**Review by Dr-Irv — APPROVED** [review]
thanks @kirill-bash

**Pull request merged**